Functional Programming/Problem8/Problem8.js

```diff
@@ -0,0 +1,51 @@
+(function () {
+  'use strict';
+
+  function linearUnfold(callback, initialValue) {
+    var state = true,
+        value = initialValue,
+        clbResult,
+        results = [];
+
+    while(state){
+      clbResult = callback(value)
+      state = clbResult.pop();
+      value = clbResult.pop();
+      if (state)
+        results.push(value);
+    }
+
+    return results;
+  }
+
+  function inject(array, callback, initValue) {
+    var memo = initValue;
+    if (memo == undefined)
+      memo = array.shift();
+ 
+    for (var index = 0; index < array.length; index++)
+      memo = callback(memo, array[index], index, array);
+
+    return memo;
+  }  
+
+  function randomizedArray(seed) {
+    var counter = 0,
+        results = [];
+
+    randomizedArray = function(seed) {
```

Comment:
What are benefits from such function reassignment? The function can't be use for further random arrays generation.